backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/S3Service.java

```diff
@@ -101,4 +107,27 @@ public String generatePresignedUrl(String s3Key, String method, long partNumber,
 
     return s3Client.generatePresignedUrl(request).toString();
   }
+
+  @Override
+  public void commitMultipart(String s3Key, String uploadId, List<PartETag> partETags)
+      throws ModelDBException {
+    // Validate bucket
+    Boolean exist = doesBucketExist(bucketName);
+    if (!exist) {
+      throw new ModelDBException("Bucket does not exists", io.grpc.Status.Code.INTERNAL);
+    }
+    CompleteMultipartUploadRequest completeMultipartUploadRequest =
+        new CompleteMultipartUploadRequest(bucketName, s3Key, uploadId, partETags);
+    try {
+      CompleteMultipartUploadResult result =
+          s3Client.completeMultipartUpload(completeMultipartUploadRequest);
+      LOGGER.info("upload result: {}", result);
+    } catch (AmazonS3Exception e) {
+      if (e.getStatusCode() == HttpStatusCodes.STATUS_CODE_BAD_REQUEST) {
+        LOGGER.info("message: {} additional details: {}", e.getMessage(), e.getAdditionalDetails());
```

Comment:
This doesn't sound like an `info` to me. An actual error happened, right?

Reply:
No. That is for errors like: "Your proposed upload is smaller than the minimum allowed size".
It looks like a validation condition that can be checked only during multipart commit call.

Reply:
Ok, got it. Let's make sure we return something clear and actionable to the user. An s3 error message is not that, since it's an implementation detail of how we store data.